File: API/Features/Ships/Dtos/ShipReadDto.cs
using API.Infrastructure.Classes;

namespace API.Features.Ships {

    public class ShipReadDto {

        public int Id { get; set; }
        public string Description { get; set; }
        public string IMO { get; set; }
        public string Flag { get; set; }
        public string RegistryNo { get; set; }
        public string Manager { get; set; }
        public string ManagerInGreece { get; set; }
        public string Agent { get; set; }
        public bool IsActive { get; set; }
        public string UserId { get; set; }
        public SimpleEntity ShipOwner { get; set; }

    }

}

File: FrontEnd/src/app/features/ships/classes/dtos/ship-read-dto.ts
import { ShipOwnerAutoCompleteVM } from 'src/app/features/shipOwners/classes/view-models/shipOwner-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'

export interface ShipReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    shipOwner: ShipOwnerAutoCompleteVM
    // Fields
    description: string
    abbreviation: string
    imo: string
    flag: string
    registryNo: string
    manager: string
    managerInGreece: string
    agent: string
    isActive: boolean

}
